account/creation.go

```diff
@@ -160,3 +171,29 @@ func FromBJJPubKeyCompToHezBJJAddress(pkComp babyjub.PublicKeyComp) (string, err
 	bjjSum := append(pkComp[:], sum)
 	return "hez:" + base64.RawURLEncoding.EncodeToString(bjjSum), nil
 }
+
+// CreateHermezAuthSignature creates the hermez wallet authentication signature
+func CreateHermezAuthSignature(ethPk *ecdsa.PrivateKey, ethAccount accounts.Account, bjjPubKeyComp babyjub.PublicKeyComp, chainID int, rollupContract common.Address) (string, error) {
```

Comment:
I think it would be better to return `common.AccountCreationAuth`

Reply:
Thanks for the suggestion but I think it's not needed. The focus of the function is only to create the account creation **SIGNATURE**. I believe we can try to follow the functional paradigm and keep the methods focused and slim as much as we can.